src/app/(auth)/signup/page.js

```diff
@@ -1,14 +1,33 @@
 "use client";
 
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import { useRouter } from "next/navigation";
 import SignUpAccount from "./components/SignUpAccount";
 import SignUpDetails from "./components/SignUpDetails";
 import SignUpEmailConfirmation from "./components/SignUpEmailConfirmation";
 import SignUpChildren from "./components/SignUpChildren";
+import { AUTH_TOKEN_NAME } from "@/app/api.mjs";
 
 export default function SignUp() {
+  const router = useRouter();
   const [currentStep, setCurrentStep] = useState(0);
+
+  useEffect(() => {
+    const authToken = localStorage.getItem(AUTH_TOKEN_NAME);
+    if (authToken) {
+      router.push('/');
+    }
+    const savedStep = localStorage.getItem('signup_current_step');
```

Comment:
getItem and setItem should use same key. While having them hardcoded in 2 places you introduce a divergent change code smell. You can check shotgun strategy code smell https://refactoring.guru/smells/shotgun-surgery

It is not really critical here, and I still would like you to learn on small examples.